src/pip/_internal/models/direct_url.py

```diff
@@ -0,0 +1,261 @@
+""" PEP 610 """
```

Comment:
Possibly. I've been careful to put only (and all) the PEP 610 implementation in this file with the corresponding test, so it can be extracted easily. pip-specific stuff in in direct_url_helpers.py.